playground/worker/dep-self-reference-url-worker/index.js

```diff
@@ -0,0 +1,9 @@
+export function startWorker(handler) {
+  const worker = new Worker(new URL('./worker.js', import.meta.url), {
```

Comment:
Would it be possible to also allow `./index.js` here?  I.e. a self reference from the main script?  This is how emscripten starts its pthread workers.  All workers (including the main page) use the same script.

Reply:
I think that should work too though one caveat is that Vite needs to bundle `index.js` twice for main script and as worker script. So, potentially some code might got duplicated (but IIRC static wasm assets can be de-duped).